src/controls/control.class.ts

```diff
@@ -1,372 +1,392 @@
-     * @return {String}
-     */
-    cursorStyleHandler: function (eventData, control /* fabricObject */) {
-      return control.cursorStyle;
-    },
+  /**
+   * Returns the action name. The basic implementation just return the actionName property.
+   * @param {Event} eventData the native mouse event
+   * @param {Control} control the current control ( likely this)
+   * @param {FabricObject} object on which the control is displayed
+   * @return {String}
+   */
+  getActionName(
+    eventData: TPointerEvent,
+    control: Control,
+    fabricObject: FabricObject
+  ) {
+    return control.actionName;
+  }
 
-    /**
-     * Returns the action name. The basic implementation just return the actionName property.
-     * @param {Event} eventData the native mouse event
-     * @param {fabric.Control} control the current control ( likely this)
-     * @param {fabric.Object} object on which the control is displayed
-     * @return {String}
-     */
-    getActionName: function (eventData, control /* fabricObject */) {
-      return control.actionName;
-    },
+  /**
+   * Returns controls visibility
+   * @param {FabricObject} object on which the control is displayed
+   * @param {String} controlKey key where the control is memorized on the
+   * @return {Boolean}
+   */
+  getVisibility(fabricObject: FabricObject, controlKey: string) {
+    // @ts-expect-error TODO remove directive once fixed
+    const objectVisibility = fabricObject._controlsVisibility;
+    if (
+      objectVisibility &&
+      typeof objectVisibility[controlKey] !== 'undefined'
+    ) {
+      return objectVisibility[controlKey];
+    }
+    return this.visible;
+  }
 
-    /**
-     * Returns controls visibility
-     * @param {fabric.Object} object on which the control is displayed
-     * @param {String} controlKey key where the control is memorized on the
-     * @return {Boolean}
-     */
-    getVisibility: function (fabricObject, controlKey) {
-      var objectVisibility = fabricObject._controlsVisibility;
-      if (
-        objectVisibility &&
-        typeof objectVisibility[controlKey] !== 'undefined'
-      ) {
-        return objectVisibility[controlKey];
-      }
-      return this.visible;
-    },
+  /**
+   * Sets controls visibility
+   * @param {Boolean} visibility for the object
+   * @return {Void}
+   */
+  setVisibility(visibility: boolean, name: string, fabricObject: FabricObject) {
+    this.visible = visibility;
+  }
 
-    /**
-     * Sets controls visibility
-     * @param {Boolean} visibility for the object
-     * @return {Void}
-     */
-    setVisibility: function (visibility /* name, fabricObject */) {
-      this.visible = visibility;
-    },
+  positionHandler(
+    dim: Point,
+    finalMatrix: TMat2D,
+    fabricObject: FabricObject,
+    currentControl: Control
+  ) {
+    return new Point(this.x, this.y)
+      .multiply(dim)
+      .add(new Point(this.offsetX, this.offsetY))
+      .transform(finalMatrix);
```

Comment:
the old code was better.
This creates 5 points... the old one was really smpler 

```js
return transformPoint(
        {
          x: this.x * dim.x + this.offsetX,
          y: this.y * dim.y + this.offsetY,
        },
        finalMatrix
      );
```

Reply:
I wonder if it makes any difference...
I'll change